### PR TITLE
Clear dangling links (Issue #3516)

### PR DESCRIPTION
**Pull request by dmoody256**
Resolves Issue #3516, continuing on from PR #3534

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `src/CHANGES.txt` (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

**Comment by dmoody256** [comment]
This should be good to merge, the failures in CI where codecov failing to upload coverage report and interactive.py intermittent failure

File: SCons/Node/FSTests.py [hunk 1]
```diff
@@ -3542,6 +3542,58 @@ def __call__(self, target, source, env, executor=None):
         dir.prepare()
 
 
+if sys.platform != 'win32' and hasattr(os, 'symlink'):
+    class CleanSymlinksTestCase(_tempdirTestCase):
+
+        def test_cleans_symlinks(self):
+            """Test the prepare() method will cleanup symlinks."""
+            with open("foo", "w") as foo:
```
Comment (bdbaddog) [inline SCons/Node/FSTests.py]: Please change to use something like this (see line 1556)
`        fp = open(test.workpath('can_not_remove'))`
This creates the files in the temporary directory created to run the test.

File: SCons/Node/FSTests.py [hunk 2]
```diff
@@ -3542,6 +3542,58 @@ def __call__(self, target, source, env, executor=None):
...
+                foo.write("baz")
+
+            os.symlink("foo", "bar")
+            bar = self.fs.File("bar")
+            bar.side_effect = True
+            bar.set_state(0)
+            assert bar.exists()
```
Comment (bdbaddog) [inline SCons/Node/FSTests.py]: Please add explanitory text for the assert. What failed.

File: SCons/Node/FSTests.py [hunk 3]
```diff
@@ -3542,6 +3542,58 @@ def __call__(self, target, source, env, executor=None):
...
+                assert False, "bar should not exist"
+            except FileNotFoundError:
+                pass
+
+            os.stat("foo")
+
+
+        def test_cleans_dangling_symlinks(self):
+            """Test the prepare() method will cleanup dangling symlinks."""
+            with open("foo", "w") as foo:
```
Comment (bdbaddog) [inline SCons/Node/FSTests.py]: See comment above re self.test.workspace... for open and file io

**Comment by bdbaddog** [comment]
@dmoody256 
I resolved the above issues in a Pull request to your branch (which is the source of this pull request).
Please click merge on:
https://github.com/dmoody256/scons/pull/5

**Comment by bdbaddog** [comment]
Fixes Issue #3516 when merged

**Comment by dmoody256** [comment]
@bdbaddog thanks for the commits. Looks like sider is not seeing the decorators, I think you can force it to pass if you want.

Also I have my PRs checked as edit by maintainers so you should be able to push commits directly to my branch:
image

**Pull request merged**